index.bs

```diff
@@ -2611,46 +4248,193 @@ partial interface MLGraphBuilder {
+        The default value is [1,1].
+
+    : <dfn>autoPad</dfn>
+    ::
+        An {{MLAutoPad}} [=string=].
+        Specifies the automatic input padding options.
+        The default value is *"explicit"*, which means that the values in the {{MLPool2dOptions/padding}} array should be used for input padding.
+        When the option is set other than *"explicit"*, the values in the {{MLPool2dOptions/padding}} array are ignored.
+
+        With the *"same-upper"* option, the padding values are automatically computed such that the additional ending padding of the spatial input dimensions would allow all of the input values in the corresponding dimension to be filtered.
+
+        The *"same-lower"* option is similar but padding is applied to the beginning padding of the spatial input dimensions instead of the ending one.
+
+    : <dfn>layout</dfn>
+    ::
+        An {{MLInputOperandLayout}} [=string=].
+        Specifies the layout format of the input and output tensor as follows:
+            - **"nchw"**
+                - input tensor: *[batches, input_channels, height, width]*
+                - output tensor: *[batches, output_channels, height, width]*
+            - **"nhwc"**:
+                - input tensor: *[batches, height, width, input_channels]*
+                - output tensor: *[batches, height, width, output_channels]*
+        The default value is *"nchw"*.
+
+    : <dfn>roundingType</dfn>
+    ::
+        An {{MLRoundingType}} [=string=].
+        Specifies the rounding function used to compute the output shape.
+
+    : <dfn>outputSizes</dfn>
+    ::
+        A sequence of {{unsigned long}} of length 2.
+        Specifies the sizes of the two spacial dimensions of the output tensor. When the output sizes are explicitly specified, the {{MLPool2dOptions/roundingType}} is ignored.
+
+        If not specified, the output sizes are automatically computed.
+
+</dl>
+
+<details open>
+  <summary>
+    To <dfn for="MLGraphBuilder" data-lt="pooling-op">create pooling operation</dfn> given |op|, |input| and |options|, run the following steps:
+  </summary>
+  <div algorithm=create-pooling-operation class=algorithm-steps>
+    1. [=Assert=]: |op| is one of "averagePool2d", "l2Pool2d", "maxPool2d".
+    1. [=Assert=]: the type of |input| is {{MLOperand}}.
+    1. If the length of of |input|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} is not 4, then throw a "{{DataError}}" {{DOMException}} and stop.
+    1. If |options|.{{MLPool2dOptions/outputSizes}} [=map/exists=], or if |options|.{{MLPool2dOptions/padding}} does not [=map/exist=], set |options|.{{MLPool2dOptions/padding}} to `[0, 0, 0, 0]`.
+    1. If the length of of |padding|.{{MLOperand/[[descriptor]]}}.{{MLOperandDescriptor/dimensions}} is not 4, then throw a "{{DataError}}" {{DOMException}} and stop.
```

Comment:
I think you mean the length of `|options|.padding` instead of `|padding|.[[descriptor]]dimensions`?
And please also add validation for `|options|.windowDimensions`, and delete the `of ` before `|padding|`.